### PR TITLE
Better error messages when using interop DSL on inner-class receiver types

### PR DESCRIPTION
**Pull request by grimmerm**

### CODE REVIEW

File: truffle/com.oracle.truffle.dsl.processor/src/com/oracle/truffle/dsl/processor/InteropProcessor.java [hunk 1]
```diff
@@ -231,6 +240,21 @@ private void generateErrorClass(Element e, final String pkg, final String fqn, f
         }
     }
 
+    private static boolean isInner(AcceptMessage message) {
+        try {
+            Class<?> receiverType = message.receiverType();
+            System.out.println(receiverType.toString() + " is an inner class: " + receiverType.isMemberClass());
```
Comment (jtulach) [inline truffle/com.oracle.truffle.dsl.processor/src/com/oracle/truffle/dsl/processor/InteropProcessor.java]: Left over logging statement.

File: truffle/com.oracle.truffle.dsl.processor/src/com/oracle/truffle/dsl/processor/InteropProcessor.java [hunk 2]
```diff
@@ -231,6 +240,21 @@ private void generateErrorClass(Element e, final String pkg, final String fqn, f
...
+            return receiverType.isMemberClass();
+        } catch (MirroredTypeException mte) {
+            // wow, annotations processors are strange
+            DeclaredType type = (DeclaredType) mte.getTypeMirror();
+            TypeElement element = (TypeElement) type.asElement();
+            boolean isInner = element.getNestingKind() == NestingKind.LOCAL || element.getNestingKind() == NestingKind.MEMBER;
+            System.out.println(element.toString() + " is an inner class: " + isInner);
```
Comment (jtulach) [inline truffle/com.oracle.truffle.dsl.processor/src/com/oracle/truffle/dsl/processor/InteropProcessor.java]: Another.

File: truffle/com.oracle.truffle.dsl.processor/src/com/oracle/truffle/dsl/processor/InteropProcessor.java [hunk 3]
```diff
@@ -231,6 +240,21 @@ private void generateErrorClass(Element e, final String pkg, final String fqn, f
         }
     }
 
+    private static boolean isInner(AcceptMessage message) {
+        try {
+            Class<?> receiverType = message.receiverType();
+            System.out.println(receiverType.toString() + " is an inner class: " + receiverType.isMemberClass());
+            return receiverType.isMemberClass();
+        } catch (MirroredTypeException mte) {
+            // wow, annotations processors are strange
```
Comment (jtulach) [inline truffle/com.oracle.truffle.dsl.processor/src/com/oracle/truffle/dsl/processor/InteropProcessor.java]: Yes, they are. Maybe the comment could explain that this exception is thrown most of the time and one needs to use the mirrors to inspect the class.

**Comment by jtulach** [comment]
Please remove the `System.out.println` statements and then feel fee to integrate.

File: truffle/com.oracle.truffle.dsl.processor/src/com/oracle/truffle/dsl/processor/InteropProcessor.java (hunk 3 above)
Comment (woess) [inline truffle/com.oracle.truffle.dsl.processor/src/com/oracle/truffle/dsl/processor/InteropProcessor.java]: This is not a good comment. Better explain why/when this happens.

**Pull request merged**